src/test/java/com/github/fishio/behaviours/TestFrozenBehaviour.java

```diff
@@ -0,0 +1,45 @@
+package com.github.fishio.behaviours;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.github.fishio.Vec2d;
+
+/**
+ * Tests the FrozenBehaviour class.
+ */
+public class TestFrozenBehaviour {
+
+	private FrozenBehaviour behaviour;
+	
+	/**
+	 * Initialises the behaviour field before each test case.
+	 */
+	@Before
+	public void setUp() {
+		behaviour = new FrozenBehaviour();
+	}
+	
+	/**
+	 * Tests the getSpeedVector method.
+	 */
+	@Test
+	public void testGetSpeedVector() {
+		// The speedVector should always be 0, 0.
+		assertEquals(new Vec2d(0, 0), behaviour.getSpeedVector());
+	}
+	
+	/**
+	 * Tests the preMove method.
+	 */
+	@Test
+	public void preMove() {
+		behaviour.preMove();
+		
+		// The speedVector should not have changed as a result of calling the preMove method.
+		assertEquals(new Vec2d(0, 0), behaviour.getSpeedVector());
+	}
+	
```

Comment:
I know it's already merged, but I still have to diagree.

This test is written for the FrozenBehaviour class, but the FrozenBehaviour class does not have a "duration", it has nothing to do with time. What you're confused with is the PuFreeze (Freeze powerup) which sets the behaviour of enemyfish to the FrozenBehaviour, then after a certain duration resets it back to its old behaviour. 
Those tests can be found in the TestPuFreeze class (testStartEffect and testEndEffect). startEffect and endEffect are called at the right moment by PuFreeze's superclass: DurationPowerUp, but that is already tested!

<testing